students/Osiddiquee/lesson05/mailroom3.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+
+'''
+simple mailroom program
+'''
+
+import sys
+
+donors = [('Donor A', [50, 100, 25, 100]),
+          ('Donor B', [100, 25]),
+          ('Donor C', [300])
+          ]
+
+def list_donors():
+    for (donor, amounts) in donors:
+        print(donor)
+
+def thank_you():
+    #create list of donor names
+    donor_name = []
+    for (donor, amounts) in donors:
+        donor_name.append(donor)
+    while True:
+        print('What do you want to do?')
+        print('1. See list of donors\n'
+              '2. Add donation to list\n'
+              '3. Quit\n')
+
+        response = input('>> ')
+        if response == '1':
+            list_donors()
+        elif response == '2':
+            print('Enter a name')
+
+            name = input('>> ')
+            if name in donor_name:
+                print('Enter an amount')
+                amount = int(input('>> '))
+                try:
+                    float(amount)
+                    for (donor, amounts) in donors:
+                        if name == donor:
+                            amounts.append(amount)
+                except ValueError:
+                    print('Please enter a valid donation')
+                    return
```

Comment:
Did you mean `continue` here? `return` will break out of the loop, so no more input will be asked. Note that `continue` is not needed as well, because there's no statements that follow this line in the loop. However, it's good to have `continue` here, just in case some other code might be added outside this whole nested if structure (but inside the while loop).

Overall, it's not too desirable to have too deep nested structure. Try to separate some part and create a function for it (we call this "refactoring"). The amount input acceptance/validation part is a good choice for such refactoring.